### ISSUE TITLE
Problem in import the previous diary.

### ISSUE THREAD

**Issue by siriusy** [jrnl-org/jrnl#930]
## Support request

Hello! Thank you for reporting an issue!
If you would fill out the below points, that would make our process a whole lot easier!

* **Please tell us about your environment:**

  - Jrnl version: (run `jrnl -v`)
jrnl version v2.4
  - How you installed Jrnl
yay -S jrnl
  - Operating system [MacOS, Linux, Windows?]
Arch Linux x86_64
* **What are you trying to do?**
I am trying to import the previous encrypted diary.
* **What have you tried?**
search online to find the same problems, and use the official manual decryption python program to decrypt. but both failed.
* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, gitter, etc)
Hello everyone, I used the encrypted version of jrnl about a year ago.  serveral month later, it was uninstalled because of some software conflicts. Recently I reinstall the new version and try to import  the previous diary. It keeps showing wrong password after countless attempts. and i don't think I forgot my password.  then I try to use the official manual decryption python program to decrypt, but it says ValueError: Input strings must be a multiple of 16 in length. I have been searching online for a long time and have never found a suitable solution.  i want to find a way to decrypt my diary no matter in what method. thank you.

**Comment by wren** [comment]
Are you okay? I'm not clear whether you need help or not, since the issue keeps getting closed. Is this resolved?

**Comment by cecep2** [comment]
Hi, it appears that I'm having the exact same problem trying to decrypt or import my old journal file today. When I try importing it to jrnl2, it keeps saying that my password is wrong. When I try the Python script for manual decryption from the docs, I also get "ValueError: Input strings must be a multiple of 16 in length". @siriusy if you found a solution, could you let me know what you did?

**Comment by wren** [comment]
To clarify, this is an old jrnl file, as in a v1 jrnl file?

**Comment by cecep2** [comment]
Yes. Perhaps the file is broken: I installed jrnl 1.9.8 in a virtualenv and tried to open my journal with that, but it says the file is corrupted. However, this adds to my confusion because I didn't touch the file for a while and if my memory isn't corrupted too, everything worked the last time I used jrnl1. Not sure if this has something so to with my failed attempts to upgrade to jrnl2 or to decrypt manually. When I have the time, I will check my time machine backups for an older version. I'm on Mac OS Catalina.

**Comment by siriusy** [comment]
@wren @cecep2 It has not been resolved yet. 
As the error message says, input strings must be a multiple of 16 in length, but the encrypted string of my diary is not, so i think there is a possibility that my diary file is damaged.   But i found that the length of my new diary written by the jrnl 2.4 is also not a multiple of 16.
I also tried to downgrade the jrnl to 1.x version multi times , but as @cecep2 mentioned,  it says the file is corrupted. but i don't think so.
from the docs, I noticed the follow information 
"you can do so with any program that supports the AES algorithm in CBC. The key used for encryption is the SHA-256-hash of your password, the IV (initialisation vector) is stored in the first 16 bytes of the encrypted file. The plain text is encoded in UTF-8 and padded according to PKCS#7 before being encrypted." 
I tried to encrypt a piece of text using the same method, and then try to decrypted it. 
but also failed, it seems there is something wrong with the decoding. i am not very familiar with these algorithms, so i think there is must be something wrong with my program.This is why i closed the issue.
I also tried to crop my diary string to meet the length requirement, but i failed to decoding them too. i remember that a few month ago I successfully  decrypted my diary using the python code  provided by the docs. so i suspect that maybe the upgrade of python caused the current problem. 
That is all i have tried, and i am a little busy recently, sorry for missing replies and thank you all.

**Comment by cecep2** [comment]
OK, I kind of fixed it using an older version of my journal file found from my time machine backups which I could upgrade to jrnl2 successfully. My best guess: It seems that every backup I did after I upgraded to MacOS Catalina using a clean install is broken. Files taken from backups before that upgrade work. Either something with Catalina changed, or I accidentally already tried to open my journal file with an older jrnl2 version before with realizing and messed something up. I really don't remember, and I have no idea what really caused the issue, but apparently in my case it was really the file that was broken.

**Comment by siriusy** [comment]
OK, I got it. thank you.

**Comment by wren** [comment]
I definitely think it's the file. I haven't been able to corrupt a v1 file with v2, yet. It leaves the file untouched if it doesn't decrypt successfully.

If anyone can find a way to break the file, please let us know. But for now, we're going to leave this closed as a file problem, and not a jrnl problem.

**Comment by siriusy** [comment]
HI, the previous problem may not be the corruption of the diary file. I successfully find back my diary. Following operations are what  I did and I don’t know how it works.
1. I deleted the `~/.jrnl_config` file;
2. I created a new diary file whose name is different from the previous one;
3. Add the old diary file path in the configuration file `~/.config/jrnl/jrnl.yaml` .
`journals:
  default: /path_to_diary/new.txt
  old: /path_to_diary/olddiary.txt
`replacement may also works.
that's all.
when using  jrnl old, all my diaries come backb everything works normally and no file upgrade warning.
my jrnl version: v2.4.2
